src/main/kotlin/controllers/MainController.kt

```diff
@@ -225,4 +243,84 @@ class MainController constructor(
         worldRendererController.renderer.zLevelsSelected[z] = isSelected
         worldRendererController.renderer.isSceneUploadRequired = true
     }
+
+    private fun checkForUpdates() {
+        val now = System.currentTimeMillis() / 1000L
+        val lastChecked = configuration.getProp(SettingsController.LAST_CHECKED_FOR_UPDATES_PROP).toLongOrNull()
+
+        // see if it's been an hour since the last check
+        if (lastChecked != null && (now - lastChecked) < 3600) {
+            logger.info("Checked for updates within the past hour. Skipping check...")
+            return
+        }
+
+        try {
+            val url = URL("https://api.github.com/repos/ConnorDY/OSRS-Environment-Exporter/releases")
+
+            // create request
+            val conn = url.openConnection() as HttpsURLConnection
+            conn.requestMethod = "GET"
+            conn.setRequestProperty("Accept", "application/vnd.github.manifold-preview+json")
+            conn.useCaches = false
+            conn.doOutput = true
+
+            // capture the response
+            val inputStream = conn.inputStream
+            val bufferedReader = BufferedReader(InputStreamReader(inputStream))
+            val buffer = StringBuilder()
+
+            var line: String?
+            while (true) {
+                line = bufferedReader.readLine()
+                if (line != null) buffer.append(line)
+                else break
+            }
+            val data = buffer.toString()
```

Comment:
This whole part could also be a separate function, so that this part flows like
```kt
getGithubReleaseInfo()
// ... json parsing code
if (newerReleaseExists(currentVersion, releases)) {
    // ...
}
```